### ISSUE TITLE
Add support for babel-jest with ts-jest for angular apps testing

### ISSUE THREAD

**Issue by pumano** [nrwl/nx#4729]
## Description

Currently using `ts-jest` with `babel-jest` not supported and when I add they together I got error:

```bash
"Using babel-jest and ts-jest together is not supported. See ts-jest documentation for babel integration: https://kulshekhar.github.io/ts-jest/user/config/babelConfig"
```
first of all link is outdated, but I test it today and ts-jest perfectly works with babel-jest without any problems via `jest-preset-angular`.

All is need is add:

```javascript
  transform: {
    '^.+\\.(ts|html)$': 'ts-jest',
    '^.+\\.js$': 'babel-jest',
  },
```
and 

```javascript
  transformIgnorePatterns: [
    'node_modules/(?!@ngrx|angular2-ui-switch|ng-dynamic|@cds|lit-html)',
  ],
```
where you add all libs which should be processed from node_modules via babel-jest.

I use [vmware/clarity](https://github.com/vmware/clarity) UI library which under the hood depends on their core lib `@cds/core` and `lit-html` and that libs use ES6 for imports / exports with javascript.

I tested that solution and it works, but today I want to migrate to nx monorepo and found that error message, after commenting it and run again I found that it not processed properly using `babel-jest` via `@nrwl/jest` executor as it works with just Angular and `jest-preset-angular`

## Motivation

I want working tests with babel-jest
## Suggested Implementation

add support as it implemented out of the box via `jest-preset-angular`
## Alternate Implementations

Successfully tested with `jest-preset-angular` and plain angular 11.1:
```javascript
{
  "babel-jest": "^26.6.3",
  "jest": "^26.6.3",
  "jest-preset-angular": "^8.3.2",
  "ts-jest": "^26.5.0"
}
```

**Comment by pumano** [comment]
After testing I found solution:
If I add: 
```javascript
  moduleFileExtensions: ['js', 'json', 'html', 'ts'],
  transformIgnorePatterns: [
    'node_modules/(?!@ngrx|angular2-ui-switch|ng-dynamic|@cds|lit-html)',
  ],
  transform: {
    '^.+\\.(ts|html)$': 'ts-jest',
    '^.+\\.js$': 'babel-jest',
  }
```
it works perfectly, but `@nrwl/jest` executor throw Error and I can't run it. I fix it with corresponded pull request

**Comment by roddc** [comment]
@pumano I got an error when using babel-jest with ts-jest :`Using babel-jest and ts-jest together is not supported.`

**Comment by pumano** [comment]
@roddc I currently updated to latest release 11.5.2 and all works properly. My PR landed to 11.5.2. Try to upgrade dependency.

**Comment by roddc** [comment]
> @roddc I currently updated to latest release 11.5.1 and all works properly. My PR landed to 11.5.1. Try to upgrade dependency. Looks like you have cached dependency.

I reinstall node_modules, below code still exists in the source code of jest.impl.js file, I even create a branch new workspace, and it's still not working. 
` if (transformers.includes('babel-jest') && transformers.includes('ts-jest')) {
        throw new Error('Using babel-jest and ts-jest together is not supported.\n' +
            'See ts-jest documentation for babel integration: https://kulshekhar.github.io/ts-jest/user/config/babelConfig');
    }`

**Comment by roddc** [comment]
> @roddc I currently updated to latest release 11.5.1 and all works properly. My PR landed to 11.5.1. Try to upgrade dependency. Looks like you have cached dependency.
> @roddc I currently updated to latest release 11.5.1 and all works properly. My PR landed to 11.5.1. Try to upgrade dependency. Looks like you have cached dependency.

I reinstalled node_modules, below code still exists in the source code of jest.impl.js file, I even created a brand new workspace, and it's still not working. 
` if (transformers.includes('babel-jest') && transformers.includes('ts-jest')) {
        throw new Error('Using babel-jest and ts-jest together is not supported.\n' +
            'See ts-jest documentation for babel integration: https://kulshekhar.github.io/ts-jest/user/config/babelConfig');
    }`

**Comment by roddc** [comment]
> @roddc I currently updated to latest release 11.5.1 and all works properly. My PR landed to 11.5.1. Try to upgrade dependency. Looks like you have cached dependency.

I reinstall node_modules, below code still exists in the source code of jest.impl.js file, I even create a brand new workspace, and it's still not working. 
` if (transformers.includes('babel-jest') && transformers.includes('ts-jest')) {
        throw new Error('Using babel-jest and ts-jest together is not supported.\n' +
            'See ts-jest documentation for babel integration: https://kulshekhar.github.io/ts-jest/user/config/babelConfig');
    }`

**Comment by roddc** [comment]
> @roddc I currently updated to latest release 11.5.1 and all works properly. My PR landed to 11.5.1. Try to upgrade dependency. Looks like you have cached dependency.

I reinstalled node_modules, below code still exists in the source code of jest.impl.js file, I even created a brand new workspace, and it's still not working. 
` if (transformers.includes('babel-jest') && transformers.includes('ts-jest')) {
        throw new Error('Using babel-jest and ts-jest together is not supported.\n' +
            'See ts-jest documentation for babel integration: https://kulshekhar.github.io/ts-jest/user/config/babelConfig');
    }`

**Comment by rikriki** [comment]
@pumano  I tried your approach and like @roddc, I didn't work in my case. Do you have sample repo for it or can you please share your babel.config.js? I just currently update my nx workspace to 11.5.1 and do all the migrations setup and npm clean cache for cache dependency but still has no luck. Thanks!

**Comment by pumano** [comment]
@rikriki @roddc yes guys, I agree, I use my patched version instead of nx, waiting for 11.5.2
<img width="573" src="https://user-images.githubusercontent.com/17675259/111127514-feaf2300-8584-11eb-9681-a6aad5261d80.png">

**Comment by rikriki** [comment]
Yes you are right, I missed that part. I tried what you suggest and it works.  11.5.1 release  doesn't implement the code you suggest... I can still see the error message on jest.impl.ts of 11.5.1 release
![image](https://user-images.githubusercontent.com/1086653/111223398-7dbb5000-85b3-11eb-8469-b5dcd3264767.png)
